File: 237_Delete_Node_in_a_Linked_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} node
    # @return {void} Do not return anything, modify node in-place instead.
    def deleteNode(self, node):
        current = node
        nnode = current.next
        while(current.next != None):
            current.val = nnode.val
            pre = current
            current = nnode
            nnode = current.next
        pre.next = None

File: 237_Delete_Node_in_a_Linked_List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
void deleteNode(struct ListNode* node) {
    struct ListNode *nex = node;
    struct ListNode *pre = node;
    while(node->next != NULL){
        nex = node->next;
        node->val = nex->val;
        pre = node;
        node = nex;
    }
    pre->next = NULL;
}